test/common/network/listener_impl_test.cc

```diff
@@ -171,6 +259,402 @@ TEST_P(ListenerImplTest, UseActualDst) {
   dispatcher_.run(Event::Dispatcher::RunType::Block);
 }
 
+/**
+ * Tests UDP listener for actual destination and data.
+ */
+TEST_P(ListenerImplTest, UseActualDstUdp) {
```

Comment:
now that we have udp_listener_impl.* files, we should probably move shared test code to a common utility class and put these tests in /test/common/network/udp_listener_impl_test, yeah?